### DISCUSSION TITLE
Widgets stops working randomly.

### DISCUSSION THREAD

**Discussion by VolcanicMelodies** [baffler/Transparent-Twitch-Chat-Overlay#267] [Q&A]
Hello. I am a streamer who using this software. Twitch is fine but i am using Widgets for the Youtube chats. I am using Streamlabs chatbox. 
It is working but randomly stops working. No more i am able to see messages on my screen. I have to close and run the software again.
How can i solve this?

**Comment by finalmakerr** [comment]
Check Twitch see if the problem will remain. I use streamelements though and np using Twitch **BUT** YouTube cause issues last two months. 

YouTube live chat API changed few months ago just so you know 😉

**Reply by VolcanicMelodies** [reply]
I believe it is not about this software. I have realized that, i can always see chat messages on live (widget that shows on the screen) BUT when i add it as widget FOR MY ON-SCREEN it stops working after few minutes.
I am using Streamlabs..

**Reply by baffler** [reply]
I want to be able to reproduce this. Can you give me some exact steps? I've looked into what could be causing this, but I want to be able to reproduce it and see if I can fix it. Which chat are you using? Which streamlabs widget/widgets? Does it only take a couple minutes or more? Is it always the same amount of time? Does it only do it when something else is in the foreground like a game? Any other details you can think of.

**Reply by VolcanicMelodies** [reply]
Ah hello baffler.
When i use Youtube chat link as a widget, it works always.
But it is not working with streamlabs chatbox.

Youtube chat style and icons far better than streamlabs but I HAVE TO open youtube > unlink chat > copy chat link > delete widget > create new widget > find and paste css code EVERY TIME if i want to use youtube chat. Because youtube chat ALWAYS changes for each stream. **If i was able to know how to get permanent chat link, my issues will be solved. OR different provider except of streamlabs chatbox. **

I was sent message to streamlabs but they told me to this "Thank you very much for providing this example!
Unfortunately, this would be part of our known issue with YouTube rate-limiting our users. We are in talks with them but have not heard anything regarding a solution. 
I will ask our developers if there are any updates, but at this time, there is nothing on our end that can be done (as it is YouTube's API).
Please let me know if there is anything else that I can do for you."

Your software for the twitch and people still can use your widgets for different usage. You already did good job. 

**--- Scenario ---**

https://streamlabs.com/dashboard#/chatbox Go to this section and you are able to see "Click to Copy Widget URL" and create chatbox widget with this link. There is no exact time but i do not remember that i have used this for full 60 minutes. When i realize peope stop chatting, i am going to check my chat and they are sending messages but not appears in my screen.

**Comment by VolcanicMelodies** [comment]
Hello again bafler. It is been a long time.
At least can you teach me how can i set CSS code? At least i want to create CSS code automatically when i created new widged as default


**with this one >>>** @import url("https://fonts.googleapis.com/css?family=Arial");

@import url("https://fonts.googleapis.com/css?family=Imprima");

/* Background colors*/
body {
  overflow: hidden;
  background-color: rgba(0,0,0,0);
}
/* Transparent background. */
yt-live-chat-renderer {
  background-color: transparent !important;
}
yt-live-chat-text-message-renderer,
yt-live-chat-text-message-renderer[is-highlighted] {
  background-color: transparent !important;
}

yt-live-chat-text-message-renderer[author-type="owner"],
yt-live-chat-text-message-renderer[author-type="owner"][is-highlighted] {
  background-color: transparent !important;
}

yt-live-chat-text-message-renderer[author-type="moderator"],
yt-live-chat-text-message-renderer[author-type="moderator"][is-highlighted] {
  background-color: transparent !important;
}

yt-live-chat-text-message-renderer[author-type="member"],
yt-live-chat-text-message-renderer[author-type="member"][is-highlighted] {
  background-color: transparent !important;
}


yt-live-chat-author-chip #author-name {
  background-color: transparent !important;
}
/* Outlines */
yt-live-chat-renderer * {
  text-shadow: -2px -2px #000000,-2px -1px #000000,-2px 0px #000000,-2px 1px #000000,-2px 2px #000000,-1px -2px #000000,-1px -1px #000000,-1px 0px #000000,-1px 1px #000000,-1px 2px #000000,0px -2px #000000,0px -1px #000000,0px 0px #000000,0px 1px #000000,0px 2px #000000,1px -2px #000000,1px -1px #000000,1px 0px #000000,1px 1px #000000,1px 2px #000000,2px -2px #000000,2px -1px #000000,2px 0px #000000,2px 1px #000000,2px 2px #000000;
  font-family: "Imprima";
  font-size: 18px !important;
  line-height: normal !important;
}

yt-live-chat-text-message-renderer #content,
yt-live-chat-legacy-paid-message-renderer #content {
  overflow: initial !important;
}

/* Hide scrollbar. */
yt-live-chat-item-list-renderer #items{
  overflow: hidden !important;
}

yt-live-chat-item-list-renderer #item-scroller{
  overflow: hidden !important;
}

/* Hide header and input. */
yt-live-chat-header-renderer,
yt-live-chat-message-input-renderer {
  display: none !important;
}

/* Reduce side padding. */
yt-live-chat-text-message-renderer,
yt-live-chat-legacy-paid-message-renderer {
    padding-left: 4px !important;
  padding-right: 4px !important;
}

yt-live-chat-paid-message-renderer #header {
    padding-left: 4px !important;
  padding-right: 4px !important;
}

/* Avatars. */
yt-live-chat-text-message-renderer #author-photo,
yt-live-chat-paid-message-renderer #author-photo,
yt-live-chat-legacy-paid-message-renderer #author-photo {
  
  width: 24px !important;
  height: 24px !important;
  border-radius: 24px !important;
  margin-right: 6px !important;
}

/* Hide badges. */
yt-live-chat-text-message-renderer #author-badges {
  display: none !important;
  vertical-align: text-top !important;
}

/* Timestamps. */
yt-live-chat-text-message-renderer #timestamp {
  
  color: #999999 !important;
  font-family: "Imprima";
  font-size: 16px !important;
  line-height: 16px !important;
}

/* Badges. */
yt-live-chat-text-message-renderer #author-name[type="owner"],
yt-live-chat-text-message-renderer #author-name.owner,
yt-live-chat-text-message-renderer yt-live-chat-author-badge-renderer[type="owner"] {
  color: #ffd600 !important;
}

yt-live-chat-text-message-renderer #author-name[type="moderator"],
yt-live-chat-text-message-renderer #author-name.moderator,
yt-live-chat-text-message-renderer yt-live-chat-author-badge-renderer[type="moderator"] {
  color: #5e84f1 !important;
}

yt-live-chat-text-message-renderer #author-name[type="member"],
yt-live-chat-text-message-renderer #author-name.member,
yt-live-chat-text-message-renderer yt-live-chat-author-badge-renderer[type="member"] {
  color: #0f9d58 !important;
}

/* Channel names. */
yt-live-chat-text-message-renderer #author-name {
  color: #cccccc !important;
  font-family: "Changa One";
  font-size: 20px !important;
  line-height: 20px !important;
}

yt-live-chat-text-message-renderer #author-name::after {
  content: ":";
  margin-left: 2px;
}

/* Messages. */
yt-live-chat-text-message-renderer #message,
yt-live-chat-text-message-renderer #message * {
  color: #ffffff !important;
  font-family: "Imprima";
  font-size: 18px !important;
  line-height: normal !important;
  letter-spacing: normal !important;
}


/* SuperChat/Fan Funding Messages. */
yt-live-chat-paid-message-renderer #author-name,
yt-live-chat-paid-message-renderer #author-name *,
yt-live-chat-legacy-paid-message-renderer #event-text,
yt-live-chat-legacy-paid-message-renderer #event-text * {
  color: #ffffff !important;
  font-family: "Changa One";
  font-size: 20px !important;
  line-height: 20px !important;
}

yt-live-chat-paid-message-renderer #purchase-amount,
yt-live-chat-paid-message-renderer #purchase-amount *,
yt-live-chat-legacy-paid-message-renderer #detail-text,
yt-live-chat-legacy-paid-message-renderer #detail-text * {
  color: #ffffff !important;
  font-family: "Imprima";
  font-size: 18px !important;
  line-height: 18px !important;
}

yt-live-chat-paid-message-renderer #content,
yt-live-chat-paid-message-renderer #content * {
  color: #ffffff !important;
  font-family: "Imprima";
  font-size: 18px !important;
  line-height: 18px !important;
}

yt-live-chat-moderation-message-renderer {
  display: none !important;
}

yt-live-chat-paid-message-renderer {
  margin: 4px 0 !important;
}

yt-live-chat-legacy-paid-message-renderer {
  background-color: #0f9d58 !important;
  margin: 4px 0 !important;
}

yt-live-chat-text-message-renderer a,
yt-live-chat-legacy-paid-message-renderer a {
  text-decoration: none !important;
}

yt-live-chat-text-message-renderer[is-deleted],
yt-live-chat-legacy-paid-message-renderer[is-deleted] {
  display: none !important;
}

yt-live-chat-ticker-renderer {
  background-color: transparent !important;
  box-shadow: none !important;
}
yt-live-chat-ticker-renderer {
  display: none !important;
}


yt-live-chat-ticker-paid-message-item-renderer,
yt-live-chat-ticker-paid-message-item-renderer *,
yt-live-chat-ticker-sponsor-item-renderer,
yt-live-chat-ticker-sponsor-item-renderer * {
  color: #ffffff !important;
  font-family: "Imprima";
}

yt-live-chat-mode-change-message-renderer, 
yt-live-chat-viewer-engagement-message-renderer, 
yt-live-chat-restricted-participation-renderer {
  display: none !important;
}

yt-live-chat-banner-manager {
  display: none !important;
}
@keyframes anim {
0% { opacity: 0; }
33.33333333333333% { opacity: 1; transform: none;}
66.66666666666666% { opacity: 1; transform: none;}
100% { opacity: 0; }
}

yt-live-chat-text-message-renderer,
yt-live-chat-legacy-paid-message-renderer {
  animation: anim 90000ms;
  animation-fill-mode: forwards;
}


yt-live-chat-action-panel-renderer, 
yt-live-chat-renderer #action-panel {
  display: none !important;
}
yt-reaction-control-panel-overlay,
yt-reaction-control-panel-overlay-view-model,
yt-reaction-control-panel-view-model {
  display: none !important;
}
yt-live-chat-viewer-engagement-message-renderer {
  display: none !important;
}

INSTEAD OF THIS ONE >>> body { background-color: rgba(0, 0, 0, 0); margin: 0px auto; overflow: hidden; }
